### ISSUE TITLE
Documentation on performance tips about abstract-typed fields

### ISSUE THREAD

**Issue by HenriDeh** [JuliaLang/julia#34739]
Following [this discussion on discourse](https://discourse.julialang.org/t/are-fields-with-structtypes-to-be-avoided-as-well/34515/4), I think the documentation should be more specific about field types.

[The performance tips documentation] states that one should avoid abstract types but really it should be non-concrete types altogether that should be avoided.

**Comment by kamalojasv181** [comment]
@HenriDeh so basically we should parameterize a struct in `struct Bar{T} foo::Foo{T} end ` this way and not like this `struct Bar{T <: Foo} foo::T end` in an abstract way to improve the performance right?

**Comment by HenriDeh** [comment]
No, both your definitions are correct. In your second example T will be a concrete type so it's fine. What you should not do is `struct Bar foo::Foo end` because Foo alone is not a concrete type (but not an abstract type either, hence this issue).

**Comment by timholy** [comment]
Want to try a PR, @HenriDeh? You can edit the files in your browser, see https://help.github.com/en/github/managing-files-in-a-repository/editing-files-in-another-users-repository.

**Comment by kamalojasv181** [comment]
@timholy if no one wants, then I can contribute to it.

**Comment by HenriDeh** [comment]
I'm working on it, but your review will be welcome :)

**Comment by HenriDeh** [comment]
Done, there's a test failing about white spaces. If anyone knows what I should do about it please tell me.
